main.py

```diff
@@ -0,0 +1,68 @@
+from typing import Annotated
+
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.database import SessionLocal
+
+app = FastAPI()
+
+
+async def common_parameters(q: str | None = None, skip: int = 0, limit: int = 100):
+    return {"q": q, "skip": skip, "limit": limit}
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_books(
+        commons: Annotated[dict, Depends(common_parameters)]):
+    return commons
+
+
+@app.get("/books/{book_id}", response_model=schemas.Book)
```

Comment:
```suggestion
@app.get("/books/{book_id}/", response_model=schemas.Book)
```

the same in other similar cases